### ISSUE TITLE
Interpreting contribution of each feature on the prediction

### ISSUE THREAD

**Issue by zweicoder** [dmlc/xgboost#1873]
Is there a way to efficiently compute how each feature contributed to the final prediction? This is very useful for our business case where we have to explain what features caused the changes in the final prediction. 

I'm using the Python API and the only way I can think of right now is parsing the strings from `get_dump()` somehow

**Comment by alistairewj** [comment]
Yep see here for the Python function which returns feature importance: https://github.com/dmlc/xgboost/blob/d943720883f0e70ce1fbce809e373908b47bd506/python-package/xgboost/core.py#L1089-L1094

**Comment by spkaplan** [comment]
I have a similar question. Is there a way to determine the feature importance, not in creating the model, but instead, in making a prediction of a given input? For example, you feed an input into an already trained model, and I would like to have it return which features in this particular input example contributed the most (and how much) to it being classified the way it was.

**Comment by alistairewj** [comment]
You may want to look into the eli5 package. Here's an example with xgboost:
http://eli5.readthedocs.io/en/latest/tutorials/xgboost-titanic.html

**Comment by davidADSP** [comment]
For R users, this package may help:

https://github.com/AppliedDataSciencePartners/xgboostExplainer

Examples given in the package documentation and this blog post:

https://medium.com/applied-data-science/new-r-package-the-xgboost-explainer-51dd7d1aa211

**Comment by pmanresa** [comment]
Hi guys, sorry for writing in an old post. I was researching around a way to get feature importance of xgboost model in Scala using Spark, but I couldn't find any useful example. Concretely, I am trying to get feature importance from XGBoostModel object. Anyone could help me out with that matter? Thank you.
